src/org/labkey/test/WebDriverWrapper.java

```diff
@@ -1130,6 +1129,30 @@ public long beginAt(String relativeURL, int millis)
         }
     }
 
+    private boolean expectPageLoad(String destinationUrl)
+    {
+        String appAction = "app";
+        try
+        {
+            String currentUrl = getDriver().getCurrentUrl();
+            Crawler.ControllerActionId current = new Crawler.ControllerActionId(currentUrl);
+            Crawler.ControllerActionId destination = new Crawler.ControllerActionId(destinationUrl);
+
+            return !destinationUrl.contains("#") ||
+                    destinationUrl.contains("?#") != currentUrl.contains("?#") ||
+                    !current.getAction().equals(appAction) ||
+                    !destination.getAction().equals(appAction) ||
+                    !current.getController().equals(destination.getController()) ||
+                    !current.getFolder().equals(destination.getFolder());
```

Comment:
I don't see any obvious flaws with the comparisons, I trust you to have done the homework on the validity of each.

Just a couple of questions/comments:
If you are not paying attention it is easy to confuse `current` with `currentUrl`. Maybe change the variable names of `current` and `destination` to `currentActionId` & `destinationActionId`? Or some other value to make it obvious at a glance?

For these two checks:
```
!current.getAction().equals(appAction) ||
!destination.getAction().equals(appAction) ||
```
Would it be the same as:
`!current.getAction().equals(destination.getAction())`
Or is it only true if the action is app?